### PR TITLE
Updated app to use tracking with micro-metrics

### PR DESCRIPTION
**Pull request by ecwillis**

### CODE REVIEW

File: methods/getEmailDetails.js [hunk 1]
```diff
@@ -61,10 +63,21 @@ module.exports = function(payload, options, allDone) {
...
+      if (!rawDetails.demo) {
+        rawDetails.uuid = uuid.v4();
+      }
+
       const details = varson(rawDetails);
       done(null, details);
     },
-    validate(details, done) {
+    trackingData(details, done) {
+      if (!details.demo && settings.ENV.MICRO_METRICS_HOST) {
```
Comment (jgallen23) [inline methods/getEmailDetails.js]: we also need pagedata slug info if its used

Comment (jgallen23) [inline methods/getEmailDetails.js]: ^ as a tag

File: routes/send.js [hunk 2]
```diff
@@ -45,6 +45,12 @@ exports.send = {
       send(server, mailObj, details, done) {
         server.methods.sendMail(mailObj, details.sendIndividual, done);
       },
+      track(server, details, send, done) {
+        if (server.settings.app.ENV.MICRO_METRICS_HOST) {
```
Comment (jgallen23) [inline routes/send.js]: can we create another key for `metricsEnabled: {{ENV.MICRO_METRICS_HOST ? true: false}}` and use that everywhere?

File: routes/send.js [hunk 3]
```diff
@@ -45,6 +45,12 @@ exports.send = {
...
+          server.track('email.send', 1, { template: details.template }, { toEmail: details.to, uuid: details.uuid });
```
Comment (jgallen23) [inline routes/send.js]: need pagedata slug tags too

File: test/method.getEmailDetails.test.js [hunk 4]
```diff
@@ -31,7 +31,8 @@ tap.test('getEmailDetails - with yaml', (assert) => {
     data: {
       firstName: 'bob',
       lastName: 'smith'
-    }
+    },
+    demo: 1 // disable the tracking pixel and uuid generation
```
Comment (jgallen23) [inline test/method.getEmailDetails.test.js]: why do we need this?

Comment (ecwillis) [inline test/method.getEmailDetails.test.js]: This is "originally" a flag to disable features that you wouldn't need in either the view or in test (like tracking). I changed the flag to be more specific: `disableTracking` that you can add to the payload to override the configured settings.

File: conf/default.yaml [hunk 5]
```diff
@@ -1,13 +1,19 @@
...
+enableMetrics: "{{(ENV.MICRO_METRICS_HOST) ? true : false}}"
 plugins:
   vision:
   inert:
   hapi-pagedata:
     _enabled: "{{(ENV.PAGEDATA_HOST) ? true : false}}"
     host: "{{ENV.PAGEDATA_HOST}}"
     key: "{{ENV.PAGEDATA_KEY}}"
+  hapi-micro-metrics:
+    _enabled: "{{(ENV.MICRO_METRICS_HOST) ? true : false }}"
```
Comment (jgallen23) [inline conf/default.yaml]: use enableMetrics

File: routes/view.js [hunk 6]
```diff
@@ -37,7 +37,8 @@ exports.view = {
         const payload = {
           template: email,
           data: require(testPath),
-          to: 'test@firstandthird.com'
+          to: 'test@firstandthird.com',
+          disableTracking: true
```
Comment (jgallen23) [inline routes/view.js]: dont we need to add tests for tacking?

File: conf/dev.yaml [hunk 7]
```diff
@@ -1,2 +1,8 @@
+server:
```
Comment (jgallen23) [inline conf/dev.yaml]: whats this for?  I'm pretty sure rapptor logs everything as is

Comment (jgallen23) [inline conf/dev.yaml]: what do we need these for? rapptor should be logging everything by default

Comment (jgallen23) [inline conf/dev.yaml]: hi

File: routes/view.js (hunk 6 above)
Comment (ecwillis) [inline routes/view.js]: Added tests below.

**Pull request merged**